fleetctl/fleetctl.go

```diff
@@ -845,3 +845,28 @@ func suToGlobal(su schema.Unit) bool {
 	}
 	return u.IsGlobal()
 }
+
+// checkUnitNames sanity checks unit names
+func checkUnitNames(names []string, allowTemplate bool) error {
+	units := make([]string, 0, len(names))
```

Comment:
forgive me if I'm misunderstanding the intention, but I think this should be rewritten as 

```go
    func checkUnitNames(names []string, allowTemplate bool) error {
	for _, un := range units {
		uni := unit.NewUnitNameInfo(unitNameMangle(un))
		// snip...
```